### PR TITLE
Add v1 Backend application

### PR DESCRIPTION
**Pull request by Shreyanand**
This PR adds v1 backend application with document retrieval and answer generation. Readme has instructions on setting it up and running it. 

The vector database used is disk persisted Qdrant client and it can be substituted by the [Qdrant deployment on kubernetes](https://github.com/qdrant/qdrant-helm). 

The app has a gradio front end deployment through which it can be accessed. Also, more components can be added to the basic structure that this app provides.

### CODE REVIEW

File: backend-app/app.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from answer_generation import PROMPT_TEMPLATE, AnswerGeneration
+import argparse
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Document search and answer generation")
+    parser.add_argument("--query", required=True, help="Enter your query about the ROSA service")
+    args = parser.parse_args()
+    
+    config_path = 'config.json'
+    retrieval = DocumentRetrieval(config_path)
```
Comment (MichaelClifford) [inline backend-app/app.py:11]: On initializing this class it looks like we load all of our data into memory, and then either create an in-memory vector database or update an existing one to include the new dataset. Is that right? If so, these seem like expensive operations you'd want to do rarely. Would it be possible to break this up? Or maybe just move some of these more expensive calls out of the `__init__` function?

File: backend-app/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Config.json
+Update the config file with the path to the documents, the vector database, the embedding model name, etc.
+
+## credentials.env
+The current version uses openai model that requires the OPEN_API_KEY in a credentials.env file.  
+
+## Running
+Get candidates and final answer by running the app.py file in your terminal with the query as the command line argument.
+`app.py --query "What is ROSA?"`
```
Comment (MichaelClifford) [inline backend-app/README.md:14]:
```suggestion
`python app.py --query "What is ROSA?"`
```

File: backend-app/config.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "dataset_path": "../data/external",
+    "vdb_path": "../data/interim",
```
Comment (MichaelClifford) [inline backend-app/config.json:3]:
```suggestion
    "vdb_path": "../data/interim/collection",
```

Comment (MichaelClifford) [reply backend-app/config.json:3]: nvm. I see why this is the way it is :)

**Review by suppathak — APPROVED** [review]
/lgtm

**Pull request merged**